app/models/game.rb

```diff
@@ -56,11 +56,29 @@ def check?(is_white)
     return false if king.nil? # used for tests where there is no king generated
     pieces.where(game_id: id, is_white: !is_white).where.not(x_position: nil, y_position: nil).find_each do |piece|
       next if piece.x_position.nil? || piece.y_position.nil?
```

Comment:
Is this line redundant? We are checking for where.not(x_position: nil, y_position: nil) in the query on line 57. Wouldn't this already rule out if piece.x_position.nil? || piece.y_position.nil? Who can explain this?